src/V3LinkInc.cpp

```diff
@@ -0,0 +1,239 @@
+        if (VN_IS(innerp, VarRef)) {
+            AstNodeVarRef* vr = VN_CAST(innerp, VarRef);
+            vr->lvalue(true);
+        }
+
+        innerp->unlinkFrBack(&linker);
+
+        // Prepare an assignment
+        FileLine* fl = nodep->fileline();
+
+        AstAssign* assignp;
+        if (VN_IS(nodep, PreSub) || VN_IS(nodep, PostSub)) {
+            assignp = new AstAssign(fl, innerp, new AstSub(fl, inner_rhsp, newconstp));
+        } else {
+            assignp = new AstAssign(fl, innerp, new AstAdd(fl, inner_rhsp, newconstp));
+        }
+
+        nodep->replaceWith(assignp);
+
+        VL_DO_DANGLING(nodep->deleteTree(), nodep);
+    }
+    void prepost_stmt_visit(AstNode* nodep) {
+        if (m_unsupportedHere) {
+            nodep->v3error("Incrementation is not yet supported in this context.\n");
+        }
+
+        iterateChildren(nodep);
+
+        AstNodeVarRef* vr = VN_CAST(nodep->op1p(), VarRef);
+        AstConst* constp = VN_CAST(nodep->op2p(), Const);
+        AstNode* backp = nodep->backp();
+        AstConst* newconstp = constp->cloneTree(true);
+
+        // Prepare a temporary variable
+        FileLine* fl = backp->fileline();
+        string name = string("__Vincrement") + cvtToStr(m_modIncrementsNum++);
+        AstVar* varp = new AstVar(fl, AstVarType::BLOCKTEMP, name, vr->varp()->subDTypep());
+
+        // Declare the variable
+        insertBeforeStmt(nodep, varp);
+
+        if (VN_IS(nodep, PreAdd) || VN_IS(nodep, PreSub)) {
+            // PreAdd/PreSub operations
+            // Define what operation will we be doing
+            AstNode* operp;
+            if (VN_IS(nodep, PreSub)) {
+                operp = new AstSub(fl, new AstVarRef(fl, vr->varp(), false), newconstp);
+            } else {
+                operp = new AstAdd(fl, new AstVarRef(fl, vr->varp(), false), newconstp);
+            }
+
+            // Immediately after declaration - increment it by one
+            m_insStmtp->addHereThisAsNext(new AstAssign(fl, new AstVarRef(fl, varp, true), operp));
+
+            // Immediately after incrementing - assign it to the original variable
+            m_insStmtp->addHereThisAsNext(new AstAssign(fl, new AstVarRef(fl, vr->varp(), true),
+                                                        new AstVarRef(fl, varp, false)));
+        } else {
+            // PostAdd/PostSub operations
+            // assign the original variable to the temporary one
+            m_insStmtp->addHereThisAsNext(new AstAssign(fl, new AstVarRef(fl, varp, true),
+                                                        new AstVarRef(fl, vr->varp(), false)));
+
+            // Define what operation will we be doing
+            AstNode* operp;
+            if (VN_IS(nodep, PostSub)) {
+                operp = new AstSub(fl, new AstVarRef(fl, vr->varp(), false), newconstp);
+            } else {
+                operp = new AstAdd(fl, new AstVarRef(fl, vr->varp(), false), newconstp);
+            }
```

Comment:
Nearly duplicate code above can move both out of upper if and check for VN_IS PreSub or PostSub.